### PR TITLE
Feature #4115: default company colour setting

### PR DESCRIPTION
**Pull request by GabdaZM**
Works only in single player.

### CODE REVIEW

File: src/openttd.cpp [hunk 1]
```diff
@@ -942,6 +942,13 @@ static void MakeNewGameDone()
 	Company *c = Company::Get(COMPANY_FIRST);
 	c->settings = _settings_client.company;
 
+	/* Overwrite color from settings if needed */
+	if (c->settings.starting_colour != 16) {
+		c->colour = c->settings.starting_colour;
+		ResetCompanyLivery(c);
+		_company_colours[c->index] = (Colours)c->colour;
+    }
```
Comment (nielsmh) [inline src/openttd.cpp]: Indentation should be tabs, not spaces.

**Review by nielsmh — COMMENTED** [review]
The commit message should be "Add #4115: Default company colour setting", there isn't any "Feature" keyword.

File: src/openttd.cpp (hunk 1 above)
Comment (GabdaZM) [reply src/openttd.cpp]: Done.

File: src/table/settings.ini [hunk 2]
```diff
@@ -2755,6 +2755,18 @@ strhelp  = STR_CONFIG_SETTING_LIVERIES_HELPTEXT
 strval   = STR_CONFIG_SETTING_LIVERIES_NONE
 proc     = InvalidateCompanyLiveryWindow
 
+[SDTC_VAR]
+var      = gui.starting_colour
+type     = SLE_UINT8
+flags    = SLF_NOT_IN_SAVE | SLF_NO_NETWORK_SYNC
+guiflags = SGF_MULTISTRING
+def      = 0
```
Comment (nielsmh) [inline src/table/settings.ini]: Shouldn't the default be "random" to match current behavior?

Comment (GabdaZM) [reply src/table/settings.ini]: For a while I have tried to place the Random to the first place, but it seemed better at the end, but I forgot the default value at the first place.

**Comment by nielsmh** [comment]
Seems to work as intended, also in multiplayer games. In multiplayer, if two players have the same preferred company colour, the second to create their company just gets assigned a different colour.

Tentatively approving this, but I'd like someone else to review the UI placement and strings too, I'm not entirely sure I like it.

**Comment by andythenorth** [comment]
I'll review the UI and strings.

**Comment by andythenorth** [comment]
UI is fine.  

Strings, we could argue back and forth about 'Starting company colour' vs 'Default company colour'.

I think 'Default company colour' is a little better, but it's pretty much potato / potato.

**Review by andythenorth — APPROVED** [review]
+1 to this, the string can always be changed later as a single-line commit.  Thanks

File: src/lang/english.txt [hunk 3]
```diff
@@ -1258,6 +1259,9 @@ STR_CONFIG_SETTING_DYNAMIC_ENGINES_EXISTING_VEHICLES            :{WHITE}Changing
 STR_CONFIG_SETTING_INFRASTRUCTURE_MAINTENANCE                   :Infrastructure maintenance: {STRING2}
 STR_CONFIG_SETTING_INFRASTRUCTURE_MAINTENANCE_HELPTEXT          :When enabled, infrastructure causes maintenance costs. The cost grows over-proportional with the network size, thus affecting bigger companies more than smaller ones
 
+STR_CONFIG_SETTING_COMPANY_STARTING_COLOUR                      :Starting company colour: {STRING2}
```
Comment (planetmaker) [inline src/lang/english.txt:1262]: My comment feels a bit like bike shedding, yet: AI also start with a company colour. Maybe "Default company colour" with a help text of "Choose a company colour the player starts with by default" would make it ever so slightly clearer?

Comment (GabdaZM) [reply src/lang/english.txt:1262]: Yes, it sound better, I will change it. It is always hard to make a short and clear description, so thanks for the help.

**Comment by GabdaZM** [comment]
> Seems to work as intended, also in multiplayer games. In multiplayer, if two players have the same preferred company colour, the second to create their company just gets assigned a different colour.

In this case I think I should put in a check before assigning the colour, as I don't know at what point the color changing fails, and some variable values might be different for server and client.
Is it possible that in single player an AI company start earlier than the player company?

**Comment by planetmaker** [comment]
> > Seems to work as intended, also in multiplayer games. In multiplayer, if two players have the same preferred company colour, the second to create their company just gets assigned a different colour.

I don't see a problem there. The 2nd player simply gets another colour... whichever of two that is. In multiplayer the order of who is first will be determine by the server and is unique.

> Is it possible that in single player an AI company start earlier than the player company?

I don't think so. There always is first a player company.

**Comment by planetmaker** [comment]
Nvm my comment from yesterday. Andy is right - we can always change the wording of strings later if we feel like it.

**Pull request merged**